client/db/src/bench.rs

```diff
@@ -181,18 +183,21 @@ impl<B: BlockT> BenchmarkingState<B> {
 
 		let maybe_tracker = key_tracker.get(key);
 
-		let has_been_read = KeyTracker {
-			has_been_read: true,
-			has_been_written: false,
-		};
-
 		match maybe_tracker {
 			None => {
+				let has_been_read = KeyTracker {
+					has_been_read: true,
+					has_been_written: false,
+				};
 				key_tracker.insert(key.to_vec(), has_been_read);
 				read_write_tracker.add_read();
 			},
 			Some(tracker) => {
 				if !tracker.has_been_read {
+					let has_been_read = KeyTracker {
+						has_been_read: true,
+						has_been_written: tracker.has_been_written,
+					};
```

Comment:
This logic needed to be updated to support the scenario where we want to count a read, but ignore a write. (i.e. timestamp pallet)